rust/ql/lib/codeql/rust/internal/Type.qll

```diff
@@ -33,10 +33,20 @@ newtype TType =
   TDynTraitTypeParameter(TypeParam tp) {
     tp = any(DynTraitTypeRepr dt).getTrait().getGenericParamList().getATypeParam()
   } or
+  TImplTraitTypeParameter(ImplTraitTypeRepr implTrait, TypeParam tp) {
+    implTraitTypeParam(implTrait, _, tp)
+  } or
   TRefTypeParameter() or
   TSelfTypeParameter(Trait t) or
   TSliceTypeParameter()
 
+predicate implTraitTypeParam(ImplTraitTypeRepr implTrait, int i, TypeParam tp) {
+  tp = implTrait.getFunctionReturnPos().getGenericParamList().getTypeParam(i) and
+  // Only include type parameters of the function that occur inside the impl
+  // trait type.
+  exists(Path path | path.getParentNode*() = implTrait and resolvePath(path) = tp)
+}
```

Comment:
How does this predicate relate to `impl` traits that aren't in the return type, i.e. `impl` types in argument position?  Are they handled at all / separately?